File: check_sqlite.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const dbPath = path.resolve(__dirname, '../backend/prisma/dev.db');
const db = new sqlite3.Database(dbPath, sqlite3.OPEN_READONLY, (err) => {
    if (err) {
        console.error('Error opening database:', err.message);
        process.exit(1);
    }
    console.log('Connected to the SQLite database.');
});

db.serialize(() => {
    db.all("SELECT * FROM User", [], (err, rows) => {
        if (err) {
            console.error('Error querying users:', err.message);
            // Try lowercase 'users' just in case
            db.all("SELECT * FROM users", [], (err2, rows2) => {
                if (err2) {
                    console.error('Error querying users (lowercase):', err2.message);
                } else {
                    if (rows2.length > 0) {
                        console.log(`Found ${rows2.length} users (lowercase table):`);
                        console.log(rows2.slice(0, 3)); // Print first 3
                    } else {
                        console.log('No users found in "users" table.');
                    }
                }
            });
        } else {
            if (rows.length > 0) {
                console.log(`Found ${rows.length} users (PascalCase table):`);
                console.log(rows.slice(0, 3)); // Print first 3
            } else {
                console.log('No users found in "User" table.');
            }
        }
    });
});

db.close();

File: check_sqlite.py
import sqlite3
import os

db_path = r'd:\veg_shop\backend\prisma\dev.db'

if not os.path.exists(db_path):
    print(f"Database file not found at {db_path}")
    exit(1)

try:
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    # List tables
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
    tables = cursor.fetchall()
    print("Tables:", [t[0] for t in tables])
    
    # Check for User table
    user_table = None
    for t in tables:
        if t[0].lower() == 'user':
            user_table = t[0]
            break
        if t[0].lower() == 'users':
            user_table = t[0]
            break
            
    if user_table:
        print(f"Found user table: {user_table}")
        cursor.execute(f"SELECT COUNT(*) FROM {user_table}")
        count = cursor.fetchone()[0]
        print(f"User count: {count}")
        
        if count > 0:
            cursor.execute(f"SELECT * FROM {user_table} LIMIT 3")
            rows = cursor.fetchall()
            print("Sample users:", rows)
            
            # Get column names
            cursor.execute(f"PRAGMA table_info({user_table})")
            columns = [col[1] for col in cursor.fetchall()]
            print("Columns:", columns)
            
    else:
        print("No User/users table found.")
        
    conn.close()
    
except Exception as e:
    print(f"Error: {e}")
